### PR TITLE
Some XCC compiler fixes

### PR DESCRIPTION
**Pull request by tyrantlizard77**
This addresses some, but not all build failures when trying to build xt-sim with the XCC compiler.

### CODE REVIEW

File: samples/cpp_synchronization/src/main.cpp [hunk 1]
```diff
@@ -133,7 +133,7 @@ void coop_thread_entry(void)
 	}
 }
 
-void main(void)
+int main(void)
 {
 	struct k_timer timer;
```
Comment (andrewboie) [inline samples/cpp_synchronization/src/main.cpp]: This is allowed by the compiler, the return value is some undefined value, but the kernel ignores it anyway.

Comment (galak) [inline samples/cpp_synchronization/src/main.cpp]: seems bad form for our samples not to be clean, even if we allow.

Comment (andrewboie) [inline samples/cpp_synchronization/src/main.cpp]: I added a return 0 at the end

**Review by jukkar — COMMENTED** [review]
+1 for networking changes

**Review by nashif — APPROVED** [review]

**Pull request merged**